### PR TITLE
Enrich TTS pipeline parameters naming

### LINKED ISSUE #26369: MusicGen with TextToAudioPipeline issues
### System Info

- `transformers` version: 4.33.2
- Platform: Linux-5.15.120+-x86_64-with-glibc2.35
- Python version: 3.10.12
- Huggingface_hub version: 0.17.2
- Safetensors version: 0.3.3
- Accelerate version: not installed
- Accelerate config: not found
- PyTorch version (GPU?): 2.0.1+cu118 (True)
- Tensorflow version (GPU?): 2.13.0 (True)
- Flax version (CPU?/GPU?/TPU?): 0.7.2 (gpu)
- Jax version: 0.4.14
- JaxLib version: 0.4.14

### Who can help?

@sanchit-gandhi @ylacombe @Vaibhavs10 

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

#### Issue 1 - logging error

Load musicgen with pipeline will have a logging error

```python
from transformers import pipeline

pipe = pipeline("text-to-audio", model="facebook/musicgen-small")
data = pipe("latin salsa with violins")
```

error

```
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.10/logging/__init__.py", line 1100, in emit
    msg = self.format(record)
  File "/usr/lib/python3.10/logging/__init__.py", line 943, in format
    return fmt.format(record)
  File "/usr/lib/python3.10/logging/__init__.py", line 678, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.10/logging/__init__.py", line 368, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.10/dist-packages/colab_kernel_launcher.py", line 37, in <module>
    ColabKernelApp.launch_instance()
  File "/usr/local/lib/python3.10/dist-packages/traitlets/config/application.py", line 992, in launch_instance
    app.start()
  File "/usr/local/lib/python3.10/dist-packages/ipykernel/kernelapp.py", line 619, in start
    self.io_loop.start()
  File "/usr/local/lib/python3.10/dist-packages/tornado/platform/asyncio.py", line 195, in start
    self.asyncio_loop.run_forever()
  File "/usr/lib/python3.10/asyncio/base_events.py", line 603, in run_forever
    self._run_once()
  File "/usr/lib/python3.10/asyncio/base_events.py", line 1909, in _run_once
    handle._run()
  File "/usr/lib/python3.10/asyncio/events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/local/lib/python3.10/dist-packages/tornado/ioloop.py", line 685, in <lambda>
    lambda f: self._run_callback(functools.partial(callback, future))
  File "/usr/local/lib/python3.10/dist-packages/tornado/ioloop.py", line 738, in _run_callback
    ret = callback()
  File "/usr/local/lib/python3.10/dist-packages/tornado/gen.py", line 825, in inner
    self.ctx_run(self.run)
  File "/usr/local/lib/python3.10/dist-packages/tornado/gen.py", line 786, in run
    yielded = self.gen.send(value)
  File "/usr/local/lib/python3.10/dist-packages/ipykernel/kernelbase.py", line 361, in process_one
    yield gen.maybe_future(dispatch(*args))
  File "/usr/local/lib/python3.10/dist-packages/tornado/gen.py", line 234, in wrapper
    yielded = ctx_run(next, result)
  File "/usr/local/lib/python3.10/dist-packages/ipykernel/kernelbase.py", line 261, in dispatch_shell
    yield gen.maybe_future(handler(stream, idents, msg))
  File "/usr/local/lib/python3.10/dist-packages/tornado/gen.py", line 234, in wrapper
    yielded = ctx_run(next, result)
  File "/usr/local/lib/python3.10/dist-packages/ipykernel/kernelbase.py", line 539, in execute_request
    self.do_execute(
  File "/usr/local/lib/python3.10/dist-packages/tornado/gen.py", line 234, in wrapper
    yielded = ctx_run(next, result)
  File "/usr/local/lib/python3.10/dist-packages/ipykernel/ipkernel.py", line 302, in do_execute
    res = shell.run_cell(code, store_history=store_history, silent=silent)
  File "/usr/local/lib/python3.10/dist-packages/ipykernel/zmqshell.py", line 539, in run_cell
    return super(ZMQInteractiveShell, self).run_cell(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py", line 2975, in run_cell
    result = self._run_cell(
  File "/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py", line 3030, in _run_cell
    return runner(coro)
  File "/usr/local/lib/python3.10/dist-packages/IPython/core/async_helpers.py", line 78, in _pseudo_sync_runner
    coro.send(None)
  File "/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py", line 3257, in run_cell_async
    has_raised = await self.run_ast_nodes(code_ast.body, cell_name,
  File "/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py", line 3473, in run_ast_nodes
    if (await self.run_code(code, result,  async_=asy)):
  File "/usr/local/lib/python3.10/dist-packages/IPython/core/interactiveshell.py", line 3553, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-4-d27dc111b0bf>", line 4, in <cell line: 4>
    data = pipe("latin salsa with violins")
  File "/usr/local/lib/python3.10/dist-packages/transformers/pipelines/text_to_audio.py", line 138, in __call__
    return super().__call__(text_inputs, **forward_params)
  File "/usr/local/lib/python3.10/dist-packages/transformers/pipelines/base.py", line 1140, in __call__
    return self.run_single(inputs, preprocess_params, forward_params, postprocess_params)
  File "/usr/local/lib/python3.10/dist-packages/transformers/pipelines/base.py", line 1147, in run_single
    model_outputs = self.forward(model_inputs, **forward_params)
  File "/usr/local/lib/python3.10/dist-packages/transformers/pipelines/base.py", line 1046, in forward
    model_outputs = self._forward(model_inputs, **forward_params)
  File "/usr/local/lib/python3.10/dist-packages/transformers/pipelines/text_to_audio.py", line 112, in _forward
    output = self.model.generate(**model_inputs, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/transformers/models/musicgen/modeling_musicgen.py", line 2335, in generate
    logger.warning(
Message: 'Using the model-agnostic default `max_length` (=1500) to control the generation length.  recommend setting `max_new_tokens` to control the maximum length of the generation.'
Arguments: (<class 'UserWarning'>,)
```

#### Issue 2 - Not clear how to specify max_new_tokens

```
data = pipe("latin salsa with violins", max_new_tokens=256)
```

will give an error

```
TypeError: TextToAudioPipeline._sanitize_parameters() got an unexpected keyword argument 'max_new_tokens'
```

I would have expected the kwargs to be passed

### Expected behavior

Pipeline works for MusicGen

### PR DESCRIPTION
**Pull request by ylacombe**
# What does this PR do?

#26369 highlighted that the use of `forward_params` in the TTS pipeline was not clear enough. This PR enrich a bit the docstrings to correct this oversight.

Note that following a discussion with @Vaibhavs10, @sanchit-gandhi and @Narsil in the same issue, I came to the conclusion that:
- adding `max_new_tokens` would add confusion to the pipeline usage, since generative and non-generative models coexist
- in the same manner, I believe that renaming `forward_params` to `generate_params` would be equally as confusing

As @Narsil [noted](https://github.com/huggingface/transformers/issues/26369#issuecomment-1736275171), an user that would like to use advanced parameter controls would probably use the classic transformers usage (tokenizer + model + etc).

Of course, I'm open to discussion and modification of current behavior if this problem recurs in the future.

Fixes #26369

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).

## Who can review?
Hey @osanseviero, @Narsil and @ArthurZucker ! Do you feel like this resolve the issue for now?

### CODE REVIEW

File: src/transformers/pipelines/text_to_audio.py
Comment (sanchit-gandhi) [inline src/transformers/pipelines/text_to_audio.py:65]: Do you think it would help for consistency allowing the pipeline to accept either `forward_params` or `generate_kwargs`? If `forward_params`, then it gets passed to `.forward` or `.generate` (whichever one the model does). If `generate_kwargs`, it only gets passed if the model generates with `.generate`. IMO this would help with consistency across pipelines

Comment (ylacombe) [reply src/transformers/pipelines/text_to_audio.py:65]: I suppose this would make sense in terms of consistency, but not in terms of simplicity, as it might be confusing for users to have two dictionaries doing the same thing. Anyway, I'm happy with your solution, since users can still refer to the documentation in case of confusion.

WDYT @ArthurZucker and @Narsil ? 

I'll open a PR once we agree on that!

Comment (ArthurZucker) [reply src/transformers/pipelines/text_to_audio.py:65]: more aligned with sanchit here, generation params should be passed to `generate_kwargs` for sure and not as `forward_params` it's confusing and not consistent with our API

**Comment by Vaibhavs10** [comment]
Gentle ping here @ylacombe @sanchit-gandhi ?
I'd like to promote more complex operations and the ability to play around with pipelines for TTA/S. Would be cool to be able to do it via `generate_kwargs` to showcase similarity across various pipeline usage.

**Comment by Vaibhavs10** [comment]
Gentle ping here @ylacombe @sanchit-gandhi ?
I'd like to promote more complex operations and the ability to play around with pipelines for TTA/S. Would be cool to be able to do it via `generate_kwargs` to showcase similarity across various pipeline usage.

File: src/transformers/pipelines/text_to_audio.py
Comment (sanchit-gandhi) [inline src/transformers/pipelines/text_to_audio.py:146]: Should we error out if `generate_kwargs` are passed but the model doesn't generate? We could then nudge the user to use `forward_kwargs` if using a forward-only model in the error message. WDYT?

File: tests/pipelines/test_pipelines_text_to_audio.py
Comment (sanchit-gandhi) [inline tests/pipelines/test_pipelines_text_to_audio.py:197]: Think it would be nice to raise an error here?

File: tests/pipelines/test_pipelines_text_to_audio.py
Comment (sanchit-gandhi) [inline tests/pipelines/test_pipelines_text_to_audio.py:227]: nit
```suggestion
```

**Review by sanchit-gandhi — APPROVED** [review]
Nice! LGTM, thanks for iterating @ylacombe! Once we get the final review from @ArthurZucker we can merge

File: src/transformers/pipelines/text_to_audio.py
Comment (sanchit-gandhi) [inline src/transformers/pipelines/text_to_audio.py:65]: Makes it a bit clearer this is the music prompt:
```suggestion
    >>> outputs = music_generator("Techno music with high melodic riffs", generate_kwargs=generate_kwargs)
```

**Comment by sanchit-gandhi** [comment]
The MusicGen TTA doctest is currently timing out (> 120s). Given we already set a low generation max length (35 tokens), I don't think we can really reduce the time for this test much further. Do you think it makes sense to switch to using the VITS model on the doctest, since it'll run in <10s?

File: tests/pipelines/test_pipelines_text_to_audio.py
Comment (ylacombe) [reply tests/pipelines/test_pipelines_text_to_audio.py:227]: I don't see any differences haha

**Review by ylacombe — COMMENTED** [review]
Thanks for the review here @sanchit-gandhi ! I don't mind throwing an error here

File: tests/pipelines/test_pipelines_text_to_audio.py
Comment (sanchit-gandhi) [reply tests/pipelines/test_pipelines_text_to_audio.py:227]: Removing the comment `# make sure nothing is done if generate_kwargs passed since not related` since `generate_kwargs` is passes here and is related

File: src/transformers/pipelines/text_to_audio.py
Comment (ArthurZucker) [inline src/transformers/pipelines/text_to_audio.py:133]:
```suggestion
        kwargs = self._ensure_tensor_on_device(forward_params, device=self.device)
        forward_params = kwargs["forward_params"]
        generate_kwargs = kwargs["generate_kwargs"]
```
we can do this only once in the code then proceed

File: tests/pipelines/test_pipelines_text_to_audio.py
Comment (ArthurZucker) [inline tests/pipelines/test_pipelines_text_to_audio.py:190]: is there any better example of forward params? 
Speaker id seems to me that it could be part of a generation config (but also is not a tensor and the pr seems to say that we could have tensors here).

**Review by ArthurZucker — APPROVED** [review]
I don't really have a strong opinion, so good for me here

File: tests/pipelines/test_pipelines_text_to_audio.py
Comment (ylacombe) [reply tests/pipelines/test_pipelines_text_to_audio.py:190]: unfortunately I can only think of Vits and `speaker_id` is basically the only optional parameter of the forward pass. BTW,  tensor params are tested by `test_conversion_additional_tensor`

**Comment by ylacombe** [comment]
Thanks for the reviews here @sanchit-gandhi and @ArthurZucker, I have updated according to your comments, and will merge once the checks are done!

**Comment by sanchit-gandhi** [comment]
Looks like the doc tests might be too slow still: https://app.circleci.com/pipelines/github/huggingface/transformers/75458/workflows/bbc191e3-5c49-410e-a720-307229af6d37/jobs/957271

Should we use a different checkpoint? https://github.com/huggingface/transformers/pull/26473#issuecomment-1759485638

**Comment by ylacombe** [comment]
Hey @sanchit-gandhi, this slips my mind, looking at this in about an hour ;)

**Comment by ylacombe** [comment]
No timing out anymore, just by syncing the branch with main! Is it okay to merge in that case @amyeroberts and @sanchit-gandhi ?

**Comment by amyeroberts** [comment]
If tests are passing and you have core maintainter approval (from @ArthurZucker here) you're good to go!

**Pull request merged**
